pipeline/dbt/models/intermediate/int__geocodages.sql

```diff
@@ -22,12 +22,27 @@ WITH adresses AS (
     SELECT * FROM {{ ref('int__union_adresses') }}
 ),
 
+communes AS (
+    SELECT * FROM {{ ref('stg_decoupage_administratif__communes') }}
+),
+
+codes_postaux AS (
+    SELECT DISTINCT
+        UNNEST(communes.codes_postaux)
+        AS code_postal
+    FROM communes
+),
+
 final AS (
     SELECT
         CAST('{{ run_started_at }}' AS TIMESTAMP) AS "geocoded_at",
         adresses._di_surrogate_id                             AS "adresse_id",
         adresses.adresse                                      AS "input_adresse",
-        adresses.code_postal                                  AS "input_code_postal",
+        CASE
+            WHEN adresses.code_postal IN (SELECT codes_postaux.code_postal FROM codes_postaux)
```

Comment:
Avec ton JOIN tu peux pas t'éviter la subquery et juste faire ton CASE sur `codes_postaux.code_postal is NULL` ?